experimenter/experimenter/experiments/tests/test_models.py

```diff
@@ -805,26 +807,65 @@ def test_targeting_without_prevent_pref_conflicts_sets_prefs(self):
     @parameterized.expand(
         [
             (application, require, exclude, expected_targeting)
-            for (application, (require, exclude, expected_targeting)) in product(
-                list(NimbusExperiment.Application),
-                [
-                    ([], [], "true"),
-                    (["foo"], [], "('foo' in enrollments)"),
-                    ([], ["bar"], "(('bar' in enrollments) == false)"),
+            for (application, (require, exclude, expected_targeting)) in [
+                *product(
+                    list(NimbusExperiment.Application),
+                    [
+                        ([], [], "true"),
+                        ([("foo", None)], [], "('foo' in enrollments)"),
+                        ([], [("bar", None)], "(('bar' in enrollments) == false)"),
+                        (
+                            [("foo", None), ("bar", None)],
+                            [("baz", None)],
+                            (
+                                "(('baz' in enrollments) == false) && "
+                                "('foo' in enrollments) && "
+                                "('bar' in enrollments)"
+                            ),
+                        ),
+                    ],
+                ),
+                *[
                     (
-                        ["foo", "bar"],
-                        ["baz"],
+                        NimbusExperiment.Application.DESKTOP,
                         (
-                            "(('baz' in enrollments) == false) && "
-                            "('foo' in enrollments) && "
-                            "('bar' in enrollments)"
+                            [("foo", "control")],
+                            [],
+                            "(enrollmentsMap['foo'] == 'control')",
+                        ),
+                    ),
+                    (
+                        NimbusExperiment.Application.DESKTOP,
+                        (
+                            [],
+                            [("foo", "control")],
+                            "((enrollmentsMap['foo'] == 'control') == false)",
                         ),
                     ),
                 ],
-            )
+                *product(
+                    [
+                        app
+                        for app in NimbusExperiment.Application
+                        if app != NimbusExperiment.Application.DESKTOP
+                    ],
+                    [
+                        (
+                            [("foo", "control")],
+                            [],
+                            "(enrollments_map['foo'] == 'control')",
+                        ),
+                        (
+                            [],
+                            [("foo", "control")],
+                            "((enrollments_map['foo'] == 'control') == false)",
+                        ),
+                    ],
+                ),
+            ]
```

Comment:
loving the tests 🎉

Reply:
Yeah that nested parameterized object is getting a bit hairy but I figured I'd just keep the structure intact and tack my cases on rather than disturb the whole thing.